docs/docs/adding-search-with-algolia.md

```diff
@@ -2,7 +2,604 @@
+  font-size: 0.6em;
+  text-align: end;
+  padding: 0;
+`
+```
+
+Now we're almost done. 2 small steps remain. First, we need to put together a hit component for every type of result we want to display. In our example, these are blog posts and pages. And second, we need to call our `Search` component somewhere on our site. Here are the post and page hit components.
+
+```jsx
+// src/components/Search/PageHit.js
+import React from 'react'
+import { Highlight, Snippet } from 'react-instantsearch-dom'
+import { Link } from 'gatsby'
+
+const PageHit = clickHandler => ({ hit }) => (
+  <div>
+    <Link to={hit.slug} onClick={clickHandler}>
+      <h3>
+        <Highlight attribute="title" hit={hit} tagName="mark" />
+      </h3>
+    </Link>
+    <Snippet attribute="excerpt" hit={hit} tagName="mark" />
+  </div>
+)
+
+export default PageHit
+```
+
+```jsx
+// src/components/Search/PostHit.js
+import React from 'react'
+import { Highlight, Snippet } from 'react-instantsearch-dom'
+import { Link } from 'gatsby'
+
+const PageHit = clickHandler => ({ hit }) => (
+import React, { Fragment } from 'react'
+import { Highlight, Snippet } from 'react-instantsearch-dom'
+import { Link } from 'gatsby'
+import { Calendar } from 'styled-icons/octicons/Calendar'
+import { Tags } from 'styled-icons/fa-solid/Tags'
+
+const PostHit = clickHandler => ({ hit }) => (
+  <div>
+    <Link to={`/blog/` + hit.slug} onClick={clickHandler}>
+      <h3>
+        <Highlight attribute="title" hit={hit} tagName="mark" />
+      </h3>
+    </Link>
+    <div>
+      <Calendar size="1em" />
+      &nbsp;
+      <Highlight attribute="date" hit={hit} tagName="mark" />
+      &emsp;
+      <Tags size="1em" />
+      &nbsp;
+      {hit.tags.map((tag, index) => (
+        <Fragment key={tag}>
+          {!!index && ', '}
+          <Link to={`blog/` + tag.toLowerCase().replace(` `, `-`)}>{tag}</Link>
+        </Fragment>
+      ))}
+    </div>
+    <Snippet attribute="excerpt" hit={hit} tagName="mark" />
+  </div>
+)
+
+export default PostHit
+```
+
+Now all we need to do is import `Search` somewhere. The obvious place is the `Header` component so let's add it there.
+
+```jsx
+// src/components/Header/index.js
+import React from 'react'
+
+import { Container, Logo } from './styles'
+import Nav from '../Nav'
+import Search from '../Search'
+
+const searchIndices = [
+  { name: `Pages`, title: `Pages`, hitComp: `PageHit` },
+  { name: `Posts`, title: `Blog Posts`, hitComp: `PostHit` },
+]
+
+const Header = ({ site, transparent }) => (
+  <Container transparent={transparent}>
+    <Logo to="/" title={site.title} rel="home" />
+    <Nav />
+    <Search collapse indices={searchIndices} />
+  </Container>
+)
+
+export default Header
+```
+
+Note that this is where we define our array of search indices and pass it as a prop to `Search`. If everything works as expected, running `gatsby develop` should now give you some instant search magic in your site's header! How cool is that! :sunglasses:
```

Comment:
I'd love to see this wrap up with a video/GIF/screenshot of the working search so people know what it should look like.

Reply:
Great idea. I added that.